packages/bls-kms-local/src/agent/BlsKeyManagementSystem.ts

```diff
@@ -0,0 +1,97 @@
+import {blsSign, generateBls12381G2KeyPair} from '@mattrglobal/node-bbs-signatures'
+import Debug from 'debug'
+
+import {IKey, ManagedKeyInfo, MinimalImportableKey, TKeyType} from '@veramo/core'
+import {AbstractPrivateKeyStore, ManagedPrivateKey} from "@veramo/key-manager";
+import {KeyManagementSystem} from "@veramo/kms-local"
+
+const debug = Debug('veramo:kms:bls:local')
+
+export class BlsKeyManagementSystem extends KeyManagementSystem {
+
+  private readonly privateKeyStore: AbstractPrivateKeyStore;
+
+  constructor(keyStore: AbstractPrivateKeyStore) {
+    super(keyStore);
+    this.privateKeyStore = keyStore;
+  }
+
+  async importKey(args: Exclude<MinimalImportableKey, 'kms'>): Promise<ManagedKeyInfo> {
+    if (args.type === <TKeyType>'Bls12381G2') {
+      if (!args.type || !args.privateKeyHex || !args.publicKeyHex) {
+        throw new Error('invalid_argument: type, publicKeyHex and privateKeyHex are required to import a key')
+      }
+      const managedKey = this.asBlsManagedKeyInfo({ alias: args.kid, privateKeyHex: args.privateKeyHex, publicKeyHex: args.publicKeyHex, type: args.type });
+      await this.privateKeyStore.import({alias: managedKey.kid, ...args})
+      debug('imported key', managedKey.type, managedKey.publicKeyHex)
+      return managedKey
+    }
+    return super.importKey(args) as Promise<ManagedKeyInfo>
+  }
+
+  async createKey({ type }: { type: TKeyType }): Promise<ManagedKeyInfo> {
+    let key: ManagedKeyInfo
+    switch (type) {
+      case <TKeyType>'Bls12381G2': {
```

Comment:
would be nicer to make an enum for 'Bls12381G2'

Reply:
TKeyType is an union enum and already contains the Bls12381G2 key type

Reply:
Fixed anyway

Reply:
the point is to have 1 place where the text is placed so that you do not get miss types

Reply:
Nah, that single point is actually this interface. Please stay as close to upstream as possible